internal/service/advancedclustertpf/data_source.go

```diff
@@ -14,6 +14,12 @@ import (
 var _ datasource.DataSource = &ds{}
 var _ datasource.DataSourceWithConfigure = &ds{}
 
+const (
```

Comment:
Are these principles documented somewhere in the repo?

> Error diagnostics principle:
Summary should be a real sentence, static, and unique string.
Detail can be more sentences with parameters.
Include the attribute name for deprecated fields

Reply:
Thank you for the comments 😄
Added this to our techincal discussion.